pyiceberg/conversions.py

```diff
@@ -540,20 +560,26 @@ def _(_: StringType, val: str) -> str:
 
 
 @from_json.register(FixedType)
-def _(t: FixedType, val: str) -> bytes:
+def _(t: FixedType, val: Union[str, bytes]) -> bytes:
     """JSON hexadecimal encoded string into bytes."""
-    b = codecs.decode(val.encode(UTF8), "hex")
+    if isinstance(val, str):
+        b = codecs.decode(val.encode(UTF8), "hex")
 
-    if len(t) != len(b):
-        raise ValueError(f"FixedType has length {len(t)}, which is different from the value: {len(b)}")
+        if len(t) != len(b):
+            raise ValueError(f"FixedType has length {len(t)}, which is different from the value: {len(b)}")
 
-    return b
+        return b
+    else:
+        return val
```

Comment:
nit: should we also check the len of the bytes type input

Reply:
Good one, thanks!